cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeJobCleaner.java

```diff
@@ -88,6 +92,19 @@ public long run(MasterEnvironmentContext context) {
                 LOG.warn("Failed to cleanup job resources for job {}. This attempt will be retried later.", jobName, e);
               }
             }
+            try {
+              coreV1Api = getCoreV1Api();
+              String configMapName = KubeTwillPreparer.CONFIGMAP_NAME_PREFIX +
+                job.getMetadata().getLabels().get(KubeTwillRunnerService.RUN_ID_LABEL);
+              coreV1Api.deleteNamespacedConfigMap(configMapName, kubeNamespace, null, null,
```

Comment:
this will only delete config maps that still have a corresponding job. Is it possible to have deleted the job but not the configmap?

Instead of individually deleting the resources, is it possible to tie them together using owner refs?

Reply:
Handled this by deleting the configmap before deleting the job. So failure to delete the configmap will result in the job not getting cleaned up. KubeJobCleaner takes care of deleting completed jobs and associated configmaps